### PR TITLE
Handling alternative directory separators

### PR DESCRIPTION
**Pull request by rcrowder**
Changes to allow for handling of alternative directory separators, such as `\\` used on win32 os.platform machines.

### CODE REVIEW

**Comment by rcrowder** [comment]
@BoltzmannBrain cc @scottpurdy
Although these changes pass the limited py.tests, and locally with other testing using PyCharm (on Windows), I'm trying further tests.
```python run.py -d numenta --detect --score --normalize``` takes quite a while to run, I'd welcome any advice on how to test this further. Preferably with command line examples.

File: nab/corpus.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def getDataFiles(self):
     dataSets = [DataFile(path) for path in filePaths if ".csv" in path]
 
     def getRelativePath(srcRoot, srcPath):
-      return srcPath[srcPath.index(srcRoot)+len(srcRoot):].strip("/")
+      return srcPath[srcPath.index(srcRoot)+len(srcRoot):].strip(os.path.sep).replace(os.path.sep, "/")
```
Comment (BoltzmannBrain) [inline nab/corpus.py]: Please wrap lines at 80 char.

File: nab/util.py [hunk 2]
```diff
@@ -173,7 +173,7 @@ def absoluteFilePaths(directory):
   for dirpath,_,filenames in os.walk(directory):
     filenames = [f for f in filenames if not f[0] == "."]
     for f in filenames:
-      yield os.path.abspath(os.path.join(dirpath, f))
+      yield os.path.abspath(os.path.normpath(os.path.join(dirpath, f)))
```
Comment (BoltzmannBrain) [inline nab/util.py]: Why is `normpath` necessary here?

**Comment by BoltzmannBrain** [comment]
@rcrowder I left a couple comments, looks good otherwise.

File: nab/util.py (hunk 2 above)
Comment (rcrowder) [inline nab/util.py]: Part of some old modifications during testing. Reverting the change back works, considering this just handling a directory and file joining.

**Comment by rcrowder** [comment]
@BoltzmannBrain I've addressed your two comments. And hope that PyCharm sorted the line wrap properly?

**Comment by BoltzmannBrain** [comment]
:+1:

**Comment by rcrowder** [comment]
Thanks. Feels good to have this working on Windows.

**Pull request merged**